ZAPD/ZTexture.cpp

```diff
@@ -69,6 +71,42 @@ void ZTexture::ParseXML(tinyxml2::XMLElement* reader)
 	width = StringHelper::StrToL(widthXml);
 	height = StringHelper::StrToL(heightXml);
 
+	const auto& texTypeSizeXml = registeredAttributes.at("TexTypeSize");
+
+	if (texTypeSizeXml.wasSet)
+	{
+		if (!StringHelper::HasOnlyDigits(texTypeSizeXml.value))
+		{
+			throw std::runtime_error(StringHelper::Sprintf(
+				"ZTexture::ParseXML: Error in %s\n"
+				"\t Value of 'TexTypeSize' attribute has non-decimal digits: '%s'.\n",
+				name.c_str(), texTypeSizeXml.value.c_str()));
+		}
+
+		uint8_t typeSize = StringHelper::StrToL(texTypeSizeXml.value);
+
+		switch (typeSize)
+		{
+		case 8:
+			texTypeSize = TextureTypeSize::Size8;
+			break;
+		case 16:
+			texTypeSize = TextureTypeSize::Size16;
+			break;
+		case 32:
+			texTypeSize = TextureTypeSize::Size32;
+			break;
+		case 64:
+			texTypeSize = TextureTypeSize::Size64;
+			break;
+		default:
+			throw std::runtime_error(
+				StringHelper::Sprintf("ZTexture::ParseXML: Error in %s\n"
+			                          "\t TexTypeSize was specified but is invalid: %d\n",
+			                          name.c_str(), typeSize));
+		}
+	}  // TODO detect if texture is not aligned and force a different type
```

Comment:
Could you also add a flag to use format-based types, so that `u32` is used for `rgba32`, `u16` for `rgba16/ia16`, and `u8` for the others? MM has been considering using types based as far as possible on the size. `-p, --pixel-widths/sizes` or something.

Reply:
Actually, this whole block should probably be its own function since you need it in the reading function as well.